Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/ServerGameNetPortal.cs

```diff
@@ -93,7 +93,10 @@ private void OnClientDisconnect(ulong clientId)
 
         public void OnClientSceneChanged(ulong clientId, int sceneIndex)
         {
-            m_ClientSceneMap[clientId] = sceneIndex;
+            if (m_Portal.NetManager.IsServer)

```

Comment:
This feels like the wrong fix, we should fix the root cause. We're in ServerGameNetPortal, how come this callback is registered/called at all on clients??

Reply:
You're right, I corrected this, preventing the callback from being registered on clients and making sure it was un-registered when exiting a game. I also found another bug caused by the same scenario and included a fix for that as well. I updated the PR description to reflect those changes